js/chatbot/src/Copy.svelte

```diff
@@ -16,9 +16,30 @@
 	}
 
 	async function handle_copy() {
+		// Navigator clipboard api needs a secure context (https)
 		if ("clipboard" in navigator) {
 			await navigator.clipboard.writeText(value);
 			copy_feedback();
+		} else {
+			// Use the 'out of viewport hidden text area' trick
+			const textArea = document.createElement("textarea");
+			textArea.value = value;
+
+			// Move textarea out of the viewport so it's not visible
+			textArea.style.position = "absolute";
+			textArea.style.left = "-999999px";
```

Comment:
What about
```js
textArea.style.display = "none";
```

---

I was just about to fix the issue of duplicated buttons. Lucky to see this pr.

Reply:
😄